src/modules/packet/Template.h

```diff
@@ -60,27 +62,53 @@ class Template
 		// Add a template field that takes data from within the packet
 		void addFieldWithOffset(const InformationElement::IeInfo& ie, uint16_t offset, uint16_t header, uint32_t validPacketClasses)
```

Comment:
I'd suggest to move the implementation of this method to Template.cpp as it has now become more complex.
I don't think the header file is any longer the place for this.

Reply:
I agree.

And while we are at it, there is quite some duplicate code in `addFieldWithOffset` and `addFieldWithoutOffset`. To reduce this duplicate code the latter function could call the former.